lib/template-mixins.js

```diff
@@ -91,7 +100,7 @@ module.exports = function (t, fields, options) {
             error: this.errors && this.errors[key],
             maxlength: maxlength(key) || extension.maxlength,
             required: extension.required !== undefined ? extension.required : true,
-            pattern: extension.pattern
+            pattern: pattern(key) || extension.pattern
```

Comment:
It feels like this is the wrong way round. If an explicit pattern is configured then I'd expect that to be used in preference to one reverse-engineered from a validator.